relay-general/src/store/normalize.rs

```diff
@@ -291,166 +158,352 @@ impl<'a> NormalizeProcessor<'a> {
             EventType::Default
         }
     }
+}
 
-    fn is_security_report(&self, event: &Event) -> bool {
-        event.csp.value().is_some()
-            || event.expectct.value().is_some()
-            || event.expectstaple.value().is_some()
-            || event.hpkp.value().is_some()
+/// Emit any breakdowns
+fn normalize_breakdowns(event: &mut Event, breakdowns_config: Option<BreakdownsConfig>) {
+    match breakdowns_config {
+        None => {}
+        Some(ref config) => breakdowns::normalize_breakdowns(event, config),
     }
+}
 
-    /// Backfills common security report attributes.
-    fn normalize_security_report(&self, event: &mut Event) {
-        if !self.is_security_report(event) {
-            // This event is not a security report, exit here.
-            return;
-        }
-
-        event.logger.get_or_insert_with(|| "csp".to_string());
+/// Ensure measurements interface is only present for transaction events
+fn normalize_measurements(event: &mut Event) {
+    if event.ty.value() != Some(&EventType::Transaction) {
+        // Only transaction events may have a measurements interface
+        event.measurements = Annotated::empty();
+    }
+}
 
-        if let Some(ref client_ip) = self.config.client_ip {
-            let user = event.user.value_mut().get_or_insert_with(User::default);
-            user.ip_address = Annotated::new(client_ip.clone());
-        }
+fn normalize_user_agent(_event: &mut Event, normalize_user_agent: Option<bool>) {
+    if normalize_user_agent.unwrap_or(false) {
+        #[cfg(feature = "uaparser")]
+        user_agent::normalize_user_agent(_event);
 
-        if let Some(ref client) = self.config.user_agent {
-            let request = event
-                .request
-                .value_mut()
-                .get_or_insert_with(Request::default);
-
-            let headers = request
-                .headers
-                .value_mut()
-                .get_or_insert_with(Headers::default);
-
-            if !headers.contains("User-Agent") {
-                headers.insert(
-                    HeaderName::new("User-Agent"),
-                    Annotated::new(HeaderValue::new(client.clone())),
-                );
-            }
-        }
+        #[cfg(not(feature = "uaparser"))]
+        panic!("relay not built with uaparser feature");
```

Comment:
This looks risky, shall we just remove the `optional` in Cargo.toml and get rid of the feature gates?

https://github.com/getsentry/relay/blob/d91984ff4327973cb4f97b943736e834e1ee2b4d/relay-general/Cargo.toml#L38